go/libraries/doltcore/schema/schema.go

```diff
@@ -233,6 +233,11 @@ func MapSchemaBasedOnTagAndName(inSch, outSch Schema) ([]int, []int, error) {
 	keyMapping := make([]int, inSch.GetPKCols().Size())
 	valMapping := make([]int, inSch.GetNonPKCols().Size())
 
+	// if inSch or outSch is empty schema. This can be from added or dropped table.
+	if (len(inSch.GetPKCols().cols) == 0 && len(inSch.GetNonPKCols().cols) == 0) || (len(outSch.GetPKCols().cols) == 0 && len(outSch.GetNonPKCols().cols) == 0) {
```

Comment:
Can this be simplified to...

```suggestion
	if len(inSch.GetAllCols().cols) == 0 || len(outSch.GetAllCols().cols) == 0 {
```